recipe/recipe.go

```diff
@@ -36,3 +42,335 @@ type instance struct {
+		return fmt.Errorf("missing mandatory fields: %v", missingFields)
+	}
+
+	if invalidFields != nil {
+		return fmt.Errorf("invalid fields: %v", invalidFields)
+	}
+
+	return nil
+
+}
+
+//ValidateUpdateInstance - checks fields of instance before updating the instance of the recipe
+func (instance *Instance) ValidateUpdateInstance() error {
+	var missingFields []string
+	var invalidFields []string
+
+	// Validation to check if at least one instance field is updated
+	allStringFieldsEmpty := instance.DatasetID == "" && instance.Title == ""
+	allRemainingFieldsNil := instance.Editions == nil && instance.CodeLists == nil
+	if allStringFieldsEmpty && allRemainingFieldsNil {
+		invalidFields = append(invalidFields, "no instance fields updates given")
+	}
+
+	if instance.DatasetID != "" {
+		invalidFields = append(invalidFields, "dataset-id cannot be changed")
+	}
+
+	if instance.Editions != nil && len(instance.Editions) > 0 {
+		for j, edition := range instance.Editions {
+			if edition == "" {
+				missingFields = append(missingFields, "editions["+strconv.Itoa(j)+"]")
+			}
+		}
+	}
+	if instance.Editions != nil && len(instance.Editions) == 0 {
+		missingFields = append(missingFields, "editions")
+	}
+
+	//When doing the update, as instance.Codelist is an array, it needs to make sure that all fields of the codelist are complete
+	//This functionality is already available in validateCodelists
+	if instance.CodeLists != nil && len(instance.CodeLists) > 0 {
+		for i, codelist := range instance.CodeLists {
+			codelistMissingFields, codelistInvalidFields := codelist.validateCodelists()
+			if len(codelistMissingFields) > 0 {
+				for _, mField := range codelistMissingFields {
+					codelistMissingFields[i] = "codelists[" + strconv.Itoa(i) + "]." + mField
+				}
+			}
+			if len(codelistInvalidFields) > 0 {
+				for _, iField := range codelistInvalidFields {
+					codelistInvalidFields[i] = "codelists[" + strconv.Itoa(i) + "]." + iField
+				}
+			}
+			missingFields = append(missingFields, codelistMissingFields...)
+			invalidFields = append(invalidFields, codelistInvalidFields...)
+		}
+	}
+
+	// If any fields missing from the codelists of the instance
+	if missingFields != nil {
+		return fmt.Errorf("missing mandatory fields: %v", missingFields)
+	}
+
+	if invalidFields != nil {
+		return fmt.Errorf("invalid fields: %v", invalidFields)
+	}
+
+	return nil
+
+}
+
+//ValidateUpdateCodeList - checks fields of codelist before updating the codelist in instance of the recipe
+func (codelist *CodeList) ValidateUpdateCodeList() error {
+	var invalidFields []string
+
+	// Validation to check if at least one codelist field is updated
+	allStringFieldsEmpty := codelist.ID == "" && codelist.Name == "" && codelist.HRef == ""
+	if allStringFieldsEmpty && codelist.IsHierarchy == nil {
+		invalidFields = append(invalidFields, "no codelist fields updates given")
+	}
+
+	if codelist.ID != "" {
+		invalidFields = append(invalidFields, "id cannot be changed")
```

Comment:
again, i think its likely we would want to be able to update the ID, and consequently _ensure_ that the href is updated correctly.